tests/integration_tests/core/test_tracker_store.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from unittest.mock import Mock, patch
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+import sqlalchemy as sa
+
+from rasa.core.tracker_store import SQLTrackerStore
+from tests.integration_tests.conftest import (
+    POSTGRES_HOST,
+    POSTGRES_PORT,
+    POSTGRES_LOGIN_DB,
+    POSTGRES_TRACKER_STORE_DB,
+)
+
+# NOTE about the timeouts in this file. We want to fail fast
+# because SQLTrackerStore tries to connect several times
+# until it works. If the timeout is hit, it probably means
+# that something is wrong in the setup of the test
+
+
+@pytest.mark.timeout(10)
+def test_sql_tracker_store_with_login_db(
+    postgres_login_db_connection: sa.engine.Connection,
+):
+    tracker_store = SQLTrackerStore(
+        dialect="postgresql",
+        host=POSTGRES_HOST,
+        port=POSTGRES_PORT,
+        db=POSTGRES_TRACKER_STORE_DB,
+        login_db=POSTGRES_LOGIN_DB,
+    )
+
+    matching_rows = (
+        postgres_login_db_connection.execution_options(isolation_level="AUTOCOMMIT")
+        .execute(
+            sa.text(
+                "SELECT 1 FROM pg_catalog.pg_database WHERE datname = :database_name"
+            ),
+            database_name=POSTGRES_TRACKER_STORE_DB,
+        )
+        .rowcount
+    )
+    assert matching_rows == 1
+    assert tracker_store.engine.url.database == POSTGRES_TRACKER_STORE_DB
+    tracker_store.engine.dispose()
+
+
+@pytest.mark.timeout(10)
+def test_sql_tracker_store_with_login_db_db_already_exists(
+    postgres_login_db_connection: sa.engine.Connection,
+):
+    postgres_login_db_connection.execution_options(
+        isolation_level="AUTOCOMMIT"
+    ).execute(f"CREATE DATABASE {POSTGRES_TRACKER_STORE_DB}")
+
+    tracker_store = SQLTrackerStore(
+        dialect="postgresql",
+        host=POSTGRES_HOST,
+        port=POSTGRES_PORT,
+        db=POSTGRES_TRACKER_STORE_DB,
+        login_db=POSTGRES_LOGIN_DB,
+    )
+
+    matching_rows = (
+        postgres_login_db_connection.execution_options(isolation_level="AUTOCOMMIT")
+        .execute(
+            sa.text(
+                "SELECT 1 FROM pg_catalog.pg_database WHERE datname = :database_name"
+            ),
+            database_name=POSTGRES_TRACKER_STORE_DB,
+        )
+        .rowcount
+    )
+    assert matching_rows == 1
+    tracker_store.engine.dispose()
+
+
+@pytest.mark.timeout(10)
+def test_sql_tracker_store_with_login_db_race_condition(
```

Comment:
These 3 `SQLTrackerStore` tests cannot be run in parallel because they rely on the state of the database. That's why they're failing in the CI. Possible solutions:
- disable parallel runs for now, and activate when we want too
- use pytest markers to identify tests that should not be ran in parallel, and run integration tests with 2 commands.

thoughts?

Reply:
2nd option is preferrable from my PoV, as we'll have a lower impact on overall test runtime + it's something we'll probably need to do eventually anyway.

Reply:
Thanks Mariana! Now you know what I was talking about 😉 

There is also a third option: rewrite the tests so that they can run in parallel. It means that the names of the database vary from a test to another. I'm not a fan of this solution because I have a feeling it could get quite messy

Reply:
I'd say we will always have some test cases that will necessitate some kind of sequential execution 🤔  (I'm thinking of the onboarding tests that I'm currently working on as an example. Since I need to set a specific onboarding state and do consequential actions in the Rasa X instance, running these tests in parallel will be near impossible and likely very messy 😬 )

Reply:
If each test runs in a separate transaction (like in my suggestion above) you should be able to run them in parallel.

Reply:
this is reasonable but [doesn't work in this case](https://github.com/RasaHQ/rasa/pull/7800#discussion_r573085200):
> CREATE DATABASE cannot run inside a transaction block

Reply:
I gave a go at the second option with 598c16ffff9